File: crates/wicked-estate-extract/tests/fixtures/ocaml/sample.ml
module Geometry = struct

  type shape =
    | Circle of float
    | Rectangle of float * float
    | Triangle of float * float * float

  let area = function
    | Circle r -> Float.pi *. r *. r
    | Rectangle (w, h) -> w *. h
    | Triangle (a, b, c) ->
        let s = (a +. b +. c) /. 2.0 in
        sqrt (s *. (s -. a) *. (s -. b) *. (s -. c))

  let perimeter = function
    | Circle r -> 2.0 *. Float.pi *. r
    | Rectangle (w, h) -> 2.0 *. (w +. h)
    | Triangle (a, b, c) -> a +. b +. c

  let describe shape =
    match shape with
    | Circle r -> Printf.sprintf "Circle(r=%.2f)" r
    | Rectangle (w, h) -> Printf.sprintf "Rectangle(%.2f x %.2f)" w h
    | Triangle (a, b, c) -> Printf.sprintf "Triangle(%.2f, %.2f, %.2f)" a b c

  let largest shapes =
    match shapes with
    | [] -> None
    | _ ->
        let by_area s = area s in
        Some (List.fold_left
          (fun acc s -> if by_area s > by_area acc then s else acc)
          (List.hd shapes)
          (List.tl shapes))

end

File: crates/wicked-estate-extract/tests/fixtures/julia/sample.jl
module Geometry

export Shape, Circle, Rectangle, Triangle, area, perimeter, describe, largest

abstract type Shape end

struct Circle <: Shape
    radius::Float64
end

struct Rectangle <: Shape
    width::Float64
    height::Float64
end

struct Triangle <: Shape
    a::Float64
    b::Float64
    c::Float64
end

"""
    area(s::Shape) -> Float64

Compute the area of a shape via multiple dispatch.
"""
function area(s::Circle)
    return π * s.radius^2
end

function area(s::Rectangle)
    return s.width * s.height
end

function area(s::Triangle)
    p = (s.a + s.b + s.c) / 2.0
    return sqrt(p * (p - s.a) * (p - s.b) * (p - s.c))
end

"""
    perimeter(s::Shape) -> Float64

Compute the perimeter of a shape via multiple dispatch.
"""
function perimeter(s::Circle)
    return 2π * s.radius
end

function perimeter(s::Rectangle)
    return 2.0 * (s.width + s.height)
end

function perimeter(s::Triangle)
    return s.a + s.b + s.c
end

"""
    describe(s::Shape) -> String

Return a human-readable description of s.
"""
function describe(s::Circle)
    return "Circle(r=$(s.radius))"
end

function describe(s::Rectangle)
    return "Rectangle($(s.width)×$(s.height))"
end

function describe(s::Triangle)
    return "Triangle($(s.a), $(s.b), $(s.c))"
end

"""
    largest(shapes) -> Shape

Return the shape with the maximum area.
"""
function largest(shapes)
    return reduce((a, b) -> area(a) >= area(b) ? a : b, shapes)
end

end # module Geometry
